### PR TITLE
Pass stringified streaming detail to Transcription docker

### PR DESCRIPTION
**Pull request by zhizhongk**
*Issue #, if available:*
Pass whole event detail to the transcription docker instead of passing few properties. Previously only few properties were passed. As more properties are added and included in streaming event, a full event detail will be useful for transcription.

*Description of changes:*
This pr is to change the properties that are passed to the application. 
The goal of this pull request:
1. Move to whole event detail from few arguments
2. Don't break the application for those who already deployed since a new docker image will be released and customer's docker will pull the new release.
3. Let customers know they need to update infras.

Change docker to parse -e for the stringified event detail. Also keep parsing for previous arguments so that it won't break for customers who already deployed this repository in their account. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,6 +1,7 @@
 plugins {
     id 'java'
     id 'com.bmuschko.docker-remote-api' version '6.4.0'
+    id 'io.freefair.lombok' version '5.0.1'
```
Comment (rajpathi) [inline build.gradle:4]: check if this is the correct lombok location https://projectlombok.org/setup/gradle

File: src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java [hunk 2]
```diff
@@ -46,31 +48,47 @@ public static void main(String[] args) {
...
-        options.addRequiredOption("i", "transactionId", true, "transaction Id. UUID");
-        options.addRequiredOption("c", "callId", true, "CallId. UUID");
-        options.addRequiredOption("s", "streamingStatus", true, "Streaming Status. i.e. STARTED, ENDED");
-        options.addRequiredOption("t", "startTime", true, "Streaming Start Time. Format: yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+        options.addOption("a", "streamARN", true, "Stream ARN." );
+        options.addOption("f", "startFragmentNumber", true, "start fragement number");
+        options.addOption("i", "transactionId", true, "transaction Id. UUID");
+        options.addOption("c", "callId", true, "CallId. UUID");
+        options.addOption("s", "streamingStatus", true, "Streaming Status. i.e. STARTED, ENDED");
+        options.addOption("t", "startTime", true, "Streaming Start Time. Format: yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
```
Comment (rajpathi) [inline src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java:56]: Do we need all these options. Just have the required option of event detail payload.

File: src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java [hunk 3]
```diff
@@ -46,31 +48,47 @@ public static void main(String[] args) {
...
+                 detail = TranscribeStreamingContext.builder()
+                        .streamArn(line.getOptionValue("a"))
+                        .startFragmentNumber(line.getOptionValue("f"))
+                        .transactionId(line.getOptionValue("i"))
+                        .callId(line.getOptionValue("c"))
+                        .streamingStatus(line.getOptionValue("s"))
+                        .startTime(line.getOptionValue("t"))
+                        .platform(TranscriptionPlatform.ECS)
+                        .build();
+                logger.warn("{} stringifiedDetail is null. Please update the infrastructure.", DOCKER_KEY_PREFIX);
```
Comment (rajpathi) [inline src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java:75]: Remove this.

File: src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java [hunk 4]
```diff
@@ -46,31 +48,47 @@ public static void main(String[] args) {
...
-                    .streamARN(line.getOptionValue("a"))
-                    .firstFragementNumber(line.getOptionValue("f"))
-                    .transactionId(line.getOptionValue("i"))
-                    .callId(line.getOptionValue("c"))
-                    .streamingStatus(line.getOptionValue("s"))
-                    .startTime(line.getOptionValue("t"))
-                    .transcriptionPlatform(TranscriptionPlatform.ECS)
-                    .build();
+
+            TranscribeStreamingContext detail;
```
Comment (rajpathi) [inline src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java:63]: Add the payload models for Streaming events and deserialize accordingly. Look at https://code.amazon.com/packages/ChimeVineCanaryTwilioListenerLambda/trees/mainline

Comment (rajpathi) [reply src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java:63]: and move the deserialization logic to TranscribeStreamingHandler

File: src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingHandler.java [hunk 5]
```diff
@@ -94,20 +94,21 @@
...
     public String handleRequest(final TranscribeStreamingContext context) {
         try {
-            final String streamARN = context.streamARN();
-            final String firstFragementNumber = context.firstFragementNumber();
-            final String transactionId = context.transactionId();
-            final String callId = context.callId();
-            final String streamingStatus = context.streamingStatus();
-            final String startTime = context.startTime();
-            final TranscriptionPlatform solutionType = context.transcriptionPlatform();
+            final String streamARN = context.getStreamArn();
```
Comment (rajpathi) [inline src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingHandler.java:97]: Do the deserialization to model from string here.

File: src/main/java/com/amazonaws/kvstranscribestreaming/TranscribeStreamingContext.java [hunk 6]
```diff
@@ -20,110 +28,24 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
 public class TranscribeStreamingContext {
```
Comment (rajpathi) [inline src/main/java/com/amazonaws/kvstranscribestreaming/TranscribeStreamingContext.java:35]: Instead use StreamingEvent payload model. Copy it from EventBridgePayload package.

File: build.gradle (hunk 1 above)
Comment (zhizhongk) [reply build.gradle:4]: Will check. This is the plugin recommended in The Lombok Gradle Plugin in https://projectlombok.org/setup/gradle

File: src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java (hunk 2 above)
Comment (zhizhongk) [reply src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java:56]: I keep these options for those who already deployed our repository. Just imagine that once a new docker image is deployed, and ecs instances will start pulling new images while their actual infrastructure is not upgraded, then customer's application will break

File: src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingHandler.java (hunk 5 above)
Comment (zhizhongk) [reply src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingHandler.java]: Done in the latest pr

File: src/main/java/com/amazonaws/kvstranscribestreaming/TranscribeStreamingContext.java (hunk 6 above)
Comment (zhizhongk) [reply src/main/java/com/amazonaws/kvstranscribestreaming/TranscribeStreamingContext.java]: done. Added streaming started event

File: src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java (hunk 4 above)
Comment (zhizhongk) [reply src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java]: Done

File: src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java (hunk 3 above)
Comment (zhizhongk) [reply src/main/java/com/amazonaws/kvstranscribestreaming/KVSTranscribeStreamingDocker.java]: done

File: src/main/java/com/amazonaws/streamingeventmodel/MediaType.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+public enum MediaType {
+    AUDIO_L16("audio/L16");
+
+    private String mediaType;
+
+    MediaType(final String mediaType) {
+        this.mediaType = mediaType;
+    }
+
+    // Added for JacksonMapper to override the serialization of enum with field failureReason instead of enum type.
```
Comment (rajpathi) [inline src/main/java/com/amazonaws/streamingeventmodel/MediaType.java:17]: remove comment

File: src/main/java/com/amazonaws/streamingeventmodel/Platform.java [hunk 8]
```diff
@@ -0,0 +1,6 @@
+package com.amazonaws.streamingeventmodel;
+
+public enum Platform {
```
Comment (rajpathi) [inline src/main/java/com/amazonaws/streamingeventmodel/Platform.java:3]: not part of streaming event model

**Comment by zhizhongk** [comment]
1. Move platform out of streamingeventmodel package.
2. Add pre-validation step to check the SQS message size.
3. Address few comments...

**Pull request merged**